src/utilities/cast.ts

```diff
@@ -1,21 +1,28 @@
-function arraysToFloat32Arrays(arrays) {
-  const result = [];
+export function arraysToFloat32Arrays(arrays: number[][]): Float32Array[] {
+  const result: Float32Array[] = [];
   for (let i = 0; i < arrays.length; i++) {
     result.push(Float32Array.from(arrays[i]));
   }
   return result;
 }
-function arrayToFloat32Arrays(array) {
-  const result = [];
+
+export function arrayToFloat32Arrays(array: number[]): Float32Array[] {
+  const result: Float32Array[] = [];
   for (let i = 0; i < array.length; i++) {
     result.push(Float32Array.from([array[i]]));
   }
   return result;
 }
-function arrayToFloat32Array(array) {
+
+export function arrayToFloat32Array(array: number[]): Float32Array {
   return Float32Array.from(array);
 }
-function objectsToFloat32Arrays(objects, table, length) {
+
+export function objectsToFloat32Arrays(
+  objects: Array<Record<string, number>>,
+  table: Record<string, number>,
+  length: number
+): Float32Array[] {
   const results = [];
```

Comment:
I think `Float32Array` constructor should be preferred here.

Reply:
The `result` variable here is a an array of `Float32Array`s,  I can make that more apparent by setting the type as `const results: Float32Array[] = []`

Does this suggestion seem correct?

Reply:
Yes, we can make it apparent by using `: Float32Array[]`.

Reply:
Fixed this in the latest commit